pkg/sdn/api/types.go

```diff
@@ -60,3 +60,33 @@ type NetNamespaceList struct {
 	unversioned.ListMeta
 	Items []NetNamespace
 }
+
+// EgressFirewallPolicy gives the type of an EgressFirewallRule
+type EgressFirewallPolicy string
+
+const (
+	EgressFirewallPolicyAllow EgressFirewallPolicy = "allow"
+	EgressFirewallPolicyDeny  EgressFirewallPolicy = "deny"
+)
+
+// EgressFirewallRule contains a single outgoing traffic firewall rule
+type EgressFirewallRule struct {
+	Policy EgressFirewallPolicy
+	From   string // Namespace name or empty
```

Comment:
To be consistent with the upstream proposals so far, we would make EgressPolicyPeer be the `To` field, not the `From` field.

And to be consistent with the upstream proposals, `From` would not exist at all: instead, we would allow creating `EgressFirewall`/`EgressNetworkPolicy`/whatever objects in any namespace, and each object's rules would apply to traffic coming from the namespace that that object is in. (They would have to still be only creatable by cluster admins though, not project admins.)

We would still need somewhere to specify the final fallback/override rules for all remaining traffic (from all namespaces) that wasn't explicitly accepted or denied by any previous rule. None of the upstream proposals so far have attempted to handle that though, so there's no precedent to follow. We could use the default namespace for that, which kind of makes sense (and kind of doesn't). ?